src/framework/auth/strategies/oauth2/oauth2-strategy.ts

```diff
@@ -280,6 +318,19 @@ export class NbOAuth2AuthStrategy extends NbAuthStrategy {
     return params;
   }
 
+  protected buildPasswordRequestData(email: string, password: string ): any {
+    const params = {
+      grant_type: this.getOption('password.grantType'),
+      email: email,
+      password: password,
+    };
+
+    Object.entries(params)
```

Comment:
let's move this part
```
Object.entries(params)
      .forEach(([key, val]) => !val && delete params[key]);```

into some `cleanParams` function, as this block is duplicated three times now :)